compiler/rustc_mir/src/transform/coverage/mod.rs

```diff
@@ -492,12 +490,10 @@ fn inject_intermediate_expression(mir_body: &mut mir::Body<'tcx>, expression: Co
 }
 
 /// Convert the Span into its file name, start line and column, and end line and column
-fn make_code_region(
-    file_name: Symbol,
-    source_file: &Lrc<SourceFile>,
-    span: Span,
-    body_span: Span,
-) -> CodeRegion {
+fn make_code_region(source_map: &SourceMap, span: Span, body_span: Span) -> CodeRegion {
+    let source_file = source_map.lookup_source_file(body_span.lo());
```

Comment:
I don't see a reason to move these two assignments into `make_code_region()`. AFAICT, the `body_span` never changed. Why incur the code of recomputing the source_file and file_name every time this is called?

Reply:
Ah, I just saw a case (in reproducing the bug mentioned in #79725) where the `span` is in a different file.

I thought I was preventing this, and in many cases it does not happen.

So we may have to do what you're doing here. But I do want to dig into this a bit more in case. I wouldn't be surprised if there are side effects because I made some assumptions that the spans would only come from the body_span.

I'll add a note once I learn more, either way.

Reply:
I did upload a PR (#79818) to ensure `Instrumentor` generates coverage for only one file. I don't know if that fixes a problem you were trying to address here, or something else. Take a look and see if there's an impact either way. Thanks!